formatter/src/formatter/element.rs

```diff
@@ -409,4 +426,69 @@ mod tests {
         </div>
         "#);
     }
+
+    // Self Closing Tag
+
+    #[test]
+    fn self_closing_void_element_no_children_separate_closing_tag() {
+        let formatted = format_element_with_self_closing_tag! { < input >< / input > };
+        insta::assert_snapshot!(formatted, @"<input/>");
```

Comment:
Oh, and personally I think it would be better to format `<input/>` to `<input />`, that is what most html/jsx formatters do. Thoughts?

```suggestion
        insta::assert_snapshot!(formatted, @"<input />");
```

Reply:
I added the space, but I did not find a way to avoid also adding it when the "/>" would be alone on a line. For example, `<input key=1 class="veryveryvery longlonglong attributesattributesattributes listlistlist"/>` would format to multiple lines with the lone "/>" with a space before. Any ideas how to manage that case ?

Reply:
I see, this is actually a bit tricky. You have to use `space()` before you end the `cbox` that groups the attributes together, as you only want to use a space if that box did not have to break into multiple lines. It is not as clean as I would like, but this should do the job:

```rust
    fn attributes(&mut self, attributes: &[NodeAttribute], trailing_space: bool) {
        match attributes {
            [] => {
                if trailing_space {
                    self.printer.nbsp();
                }
            }
            [attribute] => {
                self.printer.cbox(0);
                self.printer.nbsp();
                self.attribute(attribute);
                if trailing_space {
                    self.printer.nbsp();
                }
                self.printer.end();
            }
            _ => {
                self.printer.cbox_indent();
                self.printer.space();

                let mut iter = attributes.iter().peekable();
                while let Some(attr) = iter.next() {
                    self.attribute(attr);

                    if iter.peek().is_some() {
                        self.printer.space()
                    }
                }

                if trailing_space {
                    self.printer.space(); // <-- only results in a space if the consistent box didn't break
                } else {
                    self.printer.zerobreak();
                }
                self.printer.end_dedent();
            }
        }
    }
```

Then in the opening tag function:
```rust
       self.attributes(element.attributes(), is_self_closing);

        if is_self_closing {
            self.printer.word("/>");
        } else {
            self.printer.word(">")
        }
```

Reply:
Everything else looks great, though! This one last change, and we have nicely formatted self-closing tags and can configure its behaviour :D